File: conversions/typescript/Relation.ts
import { BaseElement } from './base_elements';

export interface Relation extends BaseElement {
  // Nature
  history?: string | null;
  impact?: string | null;
  startDate?: number | null;
  endDate?: number | null;
  debt?: number | null;
  eventsIds?: string[] | null;
  // Involves
  primaryCharacterId?: string | null;
  primaryCreatureId?: string | null;
  primaryInstitutionId?: string | null;
  secondaryCharactersIds?: string[] | null;
  secondaryCreaturesIds?: string[] | null;
  secondaryInstitutionsIds?: string[] | null;
}

File: conversions/csharp_unity/Relation.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Relation : BaseElement
{
    // Nature
    public string History;
    public string Impact;
    public int? StartDate;
    public int? EndDate;
    public int? Debt;
    public List<string> EventsIds; = new List<string>();
    // Involves
    public string PrimaryCharacterId;
    public string PrimaryCreatureId;
    public string PrimaryInstitutionId;
    public List<string> SecondaryCharactersIds; = new List<string>();
    public List<string> SecondaryCreaturesIds; = new List<string>();
    public List<string> SecondaryInstitutionsIds; = new List<string>();
}
